src/content_cao.cpp

```diff
@@ -1814,6 +1832,30 @@ bool GenericCAO::directReportPunch(v3f dir, const ItemStack *punchitem,
 	return false;
 }
 
+bool GenericCAO::isNotItemStack() const
+{
+	// We know what is needed for something *not* to be
+	// an itemstack. However, we don't know for sure
+	// whether anything *is* an itemstack.
+	// (hence the check for 'not an itemstack')
+	static const v3f zero(0,0,0);
+	return
+		false
+		// Item stack may be physical or non-physical...
+		//|| m_prop.physical
+		|| m_prop.collideWithObjects
+		|| !m_is_visible
+		|| m_prop.makes_footstep_sound
+		|| m_prop.automatic_rotate == 0
+		|| m_prop.stepheight != 0
+		|| m_prop.automatic_face_movement_dir
+		// Item stack may be falling down first
+		//|| m_velocity != zero
+		// Acceleration may be zero or non-zero
+		//|| m_acceleration != zero
+		|| m_prop.visual != "wielditem";
```

Comment:
>It would be better to check the object name (if possible) instead of trying to guess if it's an item stack or not.

I wrote this code some time ago, and at that time, I couldn't find any dependable way to determine whether it is an itemstack.  don't recall the name being available. But maybe I overlooked something, or maybe it has become available since. I'll have a look at it again.

> Negative function names are confusing. GenericCAO::isItemStack() would be a much better choice here.

They are. I considered this carefully, and given the semantics, I stand by my choice that the name should be negative. See the comment near the top of the function.

`!isItemStack()` does not mean the same as `isNotItemStack()`. There are *three* possibilties:
- certainly not an itemstack
- certainly an itemstack
- not sure.